src/Installer.php

```diff
@@ -37,41 +37,44 @@
     private $factory;
 
     /**
-     * @var NullIO
+     * @var IOFactory
      */
-    private $io;
+    private $ioFactory;
 
     /**
      * @param ManagerState $state
      * @param Downloader $downloader
      * @param Filesystem $filesystem
      * @param Factory $factory
-     * @param IOInterface $io
+     * @param IOFactory $ioFactory
      */
     public function __construct(
         ManagerState $state,
         Downloader $downloader,
         Filesystem $filesystem,
         Factory $factory,
-        IOInterface $io
+        IOFactory $ioFactory
     ) {
         $this->state      = $state;
         $this->downloader = $downloader;
         $this->filesystem = $filesystem;
         $this->factory    = $factory;
-        $this->io         = $io;
+        $this->ioFactory  = $ioFactory;
     }
 
     /**
      * @param Workshop $workshop
-     * @throws WorkshopAlreadyInstalledException
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @throws \League\Flysystem\FileExistsException
      */
-    public function installWorkshop(Workshop $workshop)
+    public function installWorkshop(Workshop $workshop, InputInterface $input, OutputInterface $output)
```

Comment:
I think maybe the factory should be in the command so this class doesn't depend on the Symfony `InputInterface` and `OutputInterface`